src/Numerics/RandomRotationMatrix.h

```diff
@@ -0,0 +1,55 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2022 QMCPACK developers.
+//
+// File developed by: Mark Dewing, mdewing@anl.gov, Argonne National Laboratory
+//
+// File created by: Mark Dewing, mdewing@anl.gov, Argonne National Laboratory
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+#ifndef QMCPLUSPLUS_RANDOM_ROTATION_MATRIX_H
+#define QMCPLUSPLUS_RANDOM_ROTATION_MATRIX_H
+
+#include "Configuration.h"
+#include "Utilities/RandomGenerator.h"
+#include <array>
+
+namespace qmcplusplus
+{
+
+template<typename T>
+inline Tensor<T, 3> getRotationMatrix(T rng1, T rng2, T rng3)
+{
+  // The angular values for a random rotation matrix
+  // phi : 0 to 2*pi
+  // psi : 0 to 2*pi
+  // cth : -1 to 1  (cth = cos(theta))
+  T phi(TWOPI * rng1);
+  T psi(TWOPI * rng2);
+  T cth(1.0 - 2 * rng3);
+  T sph(std::sin(phi)), cph(std::cos(phi));
+  T sth(std::sqrt(1.0 - cth * cth));
+  T sps(std::sin(psi)), cps(std::cos(psi));
+  // clang-format off
+  return Tensor<T,3>( cph * cth * cps - sph * sps,
+                      sph * cth * cps + cph * sps,
+                     -sth * cps,
+                     -cph * cth * sps - sph * cps,
+                     -sph * cth * sps + cph * cps,
+                      sth * sps,
+                      cph * sth,
+                      sph * sth,
+                      cth);
+  // clang-format on
+}
+
+inline QMCTraits::TensorType getRandomRotationMatrix(RandomGenerator& rng)
```

Comment:
This API needs RandomGenerator. It should be moved to QMCHamiltonian. Better to keep files under `src/Numerics` away from RNG classes.
It is also better to assign a more self-explanatory function name once being moved to a better location.

Reply:
I split up the file into RotationMatrix3D (in Numerics) and moved this function to QMCHamiltonians/RandomRotationMatrix.h